documentation/source/rescap.rst

```diff
@@ -0,0 +1,120 @@
+   :caption: rescap.sv
+   :language: systemverilog
+   :start-at: // the design-under-test
+
+
+*************
+The Testbench
+*************
+
+The testbench consists of both an :term:`HDL` part and a Python/cocotb part.
+While having :term:`HDL` as part of the testbench is not common with cocotb, it is perfectly possible.
+
+The HDL part of the Testbench
+=============================
+
+The testbench :term:`HDL` part is written in SystemVerilog and instantiates the design described above
+as ``i_rescap``.
+It also contains a probe module for analog values as instance ``i_analog_probe`` —
+imagine this being a multimeter that you quickly connect to different nodes in the design,
+measuring either voltage or current.
+
+.. literalinclude:: ../../examples/mixed_signal/hdl/tb_rescap.sv
+   :caption: tb_rescap.sv
+   :language: systemverilog
+   :start-at: import nettypes_pkg
+
+The probe module can capture voltage and current of a node specified by ``node_to_probe``
+(a string in this module containing a hierarchical path).
+The capturing occurs whenever there is an edge on the logic signals
+``probe_voltage_toggle`` or ``probe_current_toggle``.
+The captured values can be read on real-value signals ``voltage`` and ``current`` in this module.
+
+Here is the capture code for ``voltage`` with the "user-interface" highlighted:
+
+.. literalinclude:: ../../examples/mixed_signal/hdl/analog_probe_cadence.sv
+   :caption: analog_probe_cadence.sv
+   :language: systemverilog
+   :start-at: var string node_to_probe
+   :end-at: end  // probe_voltage
+   :emphasize-lines: 1-4
+   :dedent: 2
+
+.. note:: This analog probe module is currently only implemented for the
+          Cadence Incisive and Xcelium simulators (with the AMS option available).
+          Help is appreciated to provide equivalent modules for other simulators.
+
+The cocotb part of the Testbench
+================================
+
+test_rescap_minimalist
+----------------------
+
+This is a very minimalist testcase.
+To run it, call:
+
+.. code-block:: bash
+
+    make SIM=xcelium TOPLEVEL=tb_rescap MODULE=test_rescap_minimalist
+
+
+The testcase supplies ``vdd`` and measures three pairs of voltage and current at ``vout``
+(same as the ``p`` terminal of the capacitor), spaced 50 ns apart.
+
+test_rescap
+-----------
+
+This is a more advanced testcase.
+To run it, call:
+
+.. code-block:: bash
+
+    make SIM=xcelium TOPLEVEL=tb_rescap MODULE=test_rescap
+
+
+The cocotb part of the testbench provides functions to:
+
+* do the sampling of voltage and current of a given node (:meth:`~test_rescap.ResCap_TB.get_sample_data()`),
+* plot the sampled data to a file (:meth:`~test_rescap.ResCap_TB.plot_data()`).
+
+The testcase supplies first a positive voltage to the circuit at ``vdd``, followed by a negative voltage,
+thus charging the capacitor in opposite directions.
+The following graph (produced by :mod:`matplotlib`) shows the charge curve.
```

Comment:
I don't think the link to matplotlib will work, if that's what you were intending. It might be better to hyperlink to their website, github, or pypi page.